src/engine/index.ts

```diff
@@ -2,13 +2,13 @@ import createCondition from './condition';
 import { translate } from './translate';
 import { PageView, EngineCondition } from '../../types';
 
-const check = (
+const check = <T>(
   conditions: EngineCondition[],
-  pageViews: PageView[],
+  pageViews: PageView<T>[],
```

Comment:
Can you explain the use of generics here and why it is there please?

Reply:
The generic here allows anyone using the library to provide page feature getters that return something other than `string[]` or `TopicModelFeature`. The engine currently doesn't support anything besides `keywords` and `topicDist` but we would could update it